src/azure-cli/azure/cli/command_modules/synapse/tests/latest/test_synapse_scenario.py

```diff
@@ -354,6 +354,80 @@ def test_spark_session_and_statements(self, resource_group):
                      self.check('state', 'killed')
                  ])
 
+    def test_access_control(self):
+        self.kwargs.update({
+            'workspace': 'testsynapseworkspace',
+            'role': 'Sql Admin',
+            'userPrincipal': 'username@microsoft.com',
+            'servicePrincipal': 'http://username-sp'})
+
+        self.cmd(
+            'az synapse role definition list --workspace-name {workspace} ',
```

Comment:
Is this an existing workspace? if yes, please add @record_only() for the test, otherwise live test will fail.

Reply:
Yes, we test it using an existing workspace. Thanks for your comments, I will add @record_only().

Reply:
Updated.

Reply:
But usually we recommend create the resources on-demand in test instead of using existing, is there any reason that you use existing workspace for test?
Using existing resource cannot leverage our live test tool.

---
In reply to: [472887020](https://github.com/Azure/azure-cli/pull/14834#discussion_r472887020) [](ancestors = 472887020)